library/Vanilla/Utility/AttachmentValidatorInterface.php

```diff
@@ -0,0 +1,24 @@
+<?php
+/**
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license Proprietary
+ */
+
+namespace Vanilla\Utility;
+
+/**
+ * An interface for classes intending to validate a media's attachment status.
+ *
+ * @package Vanilla\Utility
+ */
+interface AttachmentValidatorInterface {
+    /**
+     * Verify the current user can attach a media item to a specific resource.
+     *
+     * @param bool $canAttach Current state of whether or not a user can attach to this resource.
```

Comment:
```suggestion
      * @param bool $canAttach Current state of whether or not a user can attach to this resource. For a handler to have no effect, it can return the current `$canAttach` state. This result will be fired through all handlers in sequence by the event manager.
```